### PR TITLE
PIMS-72 Refactoring Redux

### PR DESCRIPTION
**Pull request by Fosol**
Converted all reducers and actions to redux-toolkit slices.  Moved files into central location `/store`.  This will improve maintaining existing features and adding enhancements.

> Note I don't have time to rewrite all the various **Actions**, but I have provided a sample `/store/slices/hooks/useAccessReqeust.ts`.

## Defects Discovered

While testing I found the following, but I still need to review whether they were introduced in this PR.

- Disposal Project List View (developer warning)
- User Management Edit Page (developer warning)
- Editing Property and Saving (developer warning)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -37,9 +37,16 @@ help:
 # Docker Development
 ##############################################################################
 
-restart: | stop build up ## Restart local docker environment
+restart: ## Restart local docker environment (n=service name)
```
Comment (Fosol) [inline Makefile:40]: We can now restart a single service now

File: Makefile [hunk 2]
```diff
@@ -37,9 +37,16 @@ help:
...
+	$(info Restart local docker environment)
+	@make stop n=$(n)
+	@make up n=$(n)
 
-refresh: | down build up ## Recreates local docker environment
+refresh: ## Recreates local docker environment (n=service name)
```
Comment (Fosol) [inline Makefile:45]: We can refresh a single service now

File: frontend/src/components/common/form/FastDatePicker.test.tsx [hunk 3]
```diff
@@ -2,7 +2,7 @@ import { FastDatePicker } from './FastDatePicker';
 import React from 'react';
 import { Formik } from 'formik';
 import { noop } from 'lodash';
-import { render, screen, wait } from '@testing-library/react';
+import { render, screen, waitFor } from '@testing-library/react';
```
Comment (Fosol) [inline frontend/src/components/common/form/FastDatePicker.test.tsx:5]: Updating the react testing-library required removing any `wait` statements.  However `waitFor` and `act` don't appear to play nice together.

File: frontend/src/components/common/form/StepForm/SteppedForm.tsx [hunk 4]
```diff
@@ -169,7 +169,7 @@ export const SteppedForm = function<T extends object = {}>({
                       onAddTab && onAddTab(values.data);
                       //add a new tab to the formik tab tracker.
                       setFieldValue('tabs', [
-                        ...values.tabs,
+                        ...(values?.tabs ?? []),
```
Comment (Fosol) [inline frontend/src/components/common/form/StepForm/SteppedForm.tsx:172]: Some tests this was resulting in an `undefined`

**Comment by Fosol** [comment]
@LawrenceLau2020 This PR is now ready and will unblock other work.

**Review by LawrenceLau2020 — APPROVED** [review]

**Pull request merged**